feat: Support providing Smartypants config

narthur opened the pull request:
This pull requests adds the ability for the user to specify smartypants config when they pass it to `marked.use`.

UziTech commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "marked-smartypants",
-  "version": "1.0.5",
+  "version": "1.1.0",
```
You don't need to update this version. It will be automatically updated when we release a new version.

UziTech commented on src/index.js:
```diff
@@ -1,6 +1,6 @@
 import { smartypants } from 'smartypants';
 
-export function markedSmartypants() {
+export function markedSmartypants(config = 2) {
```
This should be an options object in case we add new options in the future.

```js
markedSmartypants({ config: 2 });
```

UziTech reviewed (changes requested):
Can you also add information to the docs about this config. And you will need to change types and type tests.

narthur commented:
@UziTech Ok, I think I've addressed all your feedback. Can you take another look? I've never used `tsd` before, so let me know if I made any mistakes there.

Thanks!

UziTech reviewed (approved):
Thanks! :100:
